### ISSUE TITLE
mse in loss_fuction.h

### ISSUE THREAD

**Issue by tak1000** [tiny-dnn/tiny-dnn#1042]
I did a simple regression (MLP) with my 2155 data sets. The training seemed successfully completed, however, when I got "get_loss" with mse, "d" seems not to be divided by the total data sets which is 2155. 

input_data:2155 lines x 6 input elements
output_data:2155 lines x 1 output element

double loss = net.get_loss<tiny_dnn::mse>(input_data, target_data)
std::cout<<"mse="<< loss <<std::endl;

---- loss_function.h ----
class mse {
 public:
  static float_t f(const vec_t &y, const vec_t &t) {
    assert(y.size() == t.size());
    float_t d{0.0};

    for (size_t i = 0; i < y.size(); ++i) d += (y[i] - t[i]) * (y[i] - t[i]);
    **[Ichi]: this calculation is right. I confirmed ("predicted value" - "target value")^2 with Excel spared sheet.**

    return d / static_cast<float_t>(y.size());
    **[Ichi]: divided by one??? When I outputted "y.size()" with std::cout, it was "1".  I'm not a skillful C++ programmer. I might make a mistake.**
  }....

**Comment by ljluestc** [comment]
```

#include "tiny_dnn/tiny_dnn.h"

using namespace tiny_dnn;
using namespace tiny_dnn::layers;

int main() {
    // Define your neural network
    network<sequential> net;

    // Load and preprocess your input and target data
    // ...

    // Perform training
    // ...

    // Calculate the MSE loss using the entire dataset
    double loss = net.get_loss<tiny_dnn::mse>(input_data, target_data);

    // Print the MSE loss
    std::cout << "MSE = " << loss << std::endl;

    return 0;
}
```
